### PR TITLE
Add LDAP authentication configuration in trino stage and prod

### PR DESCRIPTION
**Pull request by rimolive**

### CODE REVIEW

File: trino/overlays/stage/trino-coordinator-dc.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  template:
+    spec:
+      volumes:
+        - name: root-ca
+          secret:
+            secretName: root-ca
+            items:
+              - key: RH-IT-Root-CA.crt
+                path: RH-IT-Root-CA.crt
+        - name: s3-mapping
```
Comment (maulikjs) [inline trino/overlays/stage/trino-coordinator-dc.yaml]: you are referencing this file here but I don't see the file anywhere in the PR

File: trino/overlays/prod/trino-secure-route.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    role: trino-coordinator
+spec:
+  to:
+    kind: Service
+    name: trino-service
+  port:
+    targetPort: 8080
+  wildcardPolicy: None
+  tls:
+    termination: edge
```
Comment (accorvin) [inline trino/overlays/prod/trino-secure-route.yaml]: Any reason not to make this secure by default (i.e. put this in base)?

File: trino/overlays/prod/trino-coordinator-dc.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                path: log.properties
+              - key: node.properties
+                path: node.properties
+              - key: password-authenticator.properties
+                path: password-authenticator.properties
+      containers:
+        - name: trino-coordinator
+          volumeMounts:
+            - mountPath: /etc/trino/certs
+              name: root-ca
```
Comment (gmfrasca) [inline trino/overlays/prod/trino-coordinator-dc.yaml]: no file end newline :bell: :bell: :bell:

File: trino/overlays/prod/root-ca-secret.enc.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+apiVersion: v1
```
Comment (gmfrasca) [inline trino/overlays/prod/root-ca-secret.enc.yaml]: It appears this and the stage root-ca secret are the same, so should we move this to bases?

File: trino/overlays/stage/trino-coordinator-dc.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+                path: log.properties
+              - key: node.properties
+                path: node.properties
+              - key: password-authenticator.properties
+                path: password-authenticator.properties
+      containers:
+        - name: trino-coordinator
+          volumeMounts:
+            - mountPath: /etc/trino/certs
+              name: root-ca
```
Comment (gmfrasca) [inline trino/overlays/stage/trino-coordinator-dc.yaml]: no file end newline :bell: :bell: :bell:

**Review by gmfrasca — COMMENTED** [review]
/lgtm

**Review by accorvin — APPROVED** [review]

**Pull request merged**